base/src/main/java/com/smartdevicelink/managers/screen/PresentAlertOperation.java

```diff
@@ -0,0 +1,518 @@
+import com.smartdevicelink.proxy.rpc.SoftButtonCapabilities;
+import com.smartdevicelink.proxy.rpc.TTSChunk;
+import com.smartdevicelink.proxy.rpc.WindowCapability;
+import com.smartdevicelink.proxy.rpc.enums.ImageFieldName;
+import com.smartdevicelink.proxy.rpc.enums.SpeechCapabilities;
+import com.smartdevicelink.proxy.rpc.listeners.OnRPCResponseListener;
+import com.smartdevicelink.util.DebugTool;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Operation that handles uploading images and audio data needed by the alert and, once the data uploads are complete, sending the alert.
+ * 
+ * Created by Julian Kast on 12/10/20.
+ */
+public class PresentAlertOperation extends Task {
+    private static final String TAG = "PresentAlertOperation";
+    private AlertView alertView;
+    private AlertCompletionListener listener;
+    private final WeakReference<ISdl> internalInterface;
+    private final WeakReference<FileManager> fileManager;
+    WindowCapability currentWindowCapability;
+    private int cancelId;
+    private List<SpeechCapabilities> speechCapabilities;
+    boolean isAlertPresented;
+    static int SOFTBUTTON_COUNT = 4;
+
+    public PresentAlertOperation(ISdl internalInterface, AlertView alertView, WindowCapability currentCapabilities, List<SpeechCapabilities> speechCapabilities, FileManager fileManager, Integer cancelId, AlertCompletionListener listener) {
+        super("PresentAlertOperation");
+        this.internalInterface = new WeakReference<>(internalInterface);
+        this.fileManager = new WeakReference<>(fileManager);
+        this.currentWindowCapability = currentCapabilities;
+        this.speechCapabilities = speechCapabilities;
+        this.alertView = alertView.clone();
+        this.listener = listener;
+        this.cancelId = cancelId;
+        this.isAlertPresented = false;
+
+        this.alertView.canceledListener = new AlertCanceledListener() {
+            @Override
+            public void onAlertCanceled() {
+                cancelAlert();
+            }
+        };
+        alertView.canceledListener = this.alertView.canceledListener;
+    }
+
+    @Override
+    public void onExecute() {
+        DebugTool.logInfo(TAG, "Alert Operation: Executing present Alert operation");
+        start();
+    }
+
+    private void start() {
+        if (getState() == Task.CANCELED) {
+            finishOperation(false, null);
+            return;
+        }
+        if (!isValidAlertViewData(alertView)) {
+            if (alertView.getAudio() != null && alertView.getAudio().getAudioData().size() > 0) {
+                DebugTool.logError(TAG, "The module does not support the use of only audio file data in an alert. " +
+                        "The alert has no data and can not be sent to the module. " +
+                        "The use of audio file data in an alert is only supported on modules supporting RPC Spec v5.0 or newer");
```

Comment:
Do we need to add this check? I dont think it is actually being made

Reply:
The check comes from if isValidAlertViewData() is false and there is audio set, at this point we can assume that the audio is an audio file and the module does not support it and that's why isValidAlertViewData came back false

Reply:
Good catch, for this I had to fix isValidAlertViewData(alertView) to check for audio Files, so if not isValidAlertViewData and there is audio in the AlertView, then this is the case.